shipment_advice_planner/wizards/shipment_advice_planner.py

```diff
@@ -0,0 +1,152 @@
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import Command, _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class ShipmentAdvicePlanner(models.TransientModel):
+    _name = "shipment.advice.planner"
+    _description = "Shipment Advice Planner"
+
+    picking_to_plan_ids = fields.Many2many(
+        comodel_name="stock.picking",
+        string="Pickings to plan",
+        required=True,
+        domain='[("can_be_planned_in_shipment_advice", "=", True),'
+        '("picking_type_id.warehouse_id", "=?", warehouse_id),]',
+        compute="_compute_picking_to_plan_ids",
+        store=True,
+        readonly=False,
+    )
+    shipment_planning_method = fields.Selection(
+        selection=[("simple", "Simple")], required=True, default="simple"
+    )
+    warehouse_id = fields.Many2one(comodel_name="stock.warehouse")
+    dock_id = fields.Many2one(
+        comodel_name="stock.dock", domain='[("warehouse_id", "=", warehouse_id)]'
+    )
+
+    @api.constrains("warehouse_id", "dock_id", "picking_to_plan_ids")
+    def _check_warehouse(self):
+        for rec in self:
+            if not rec.warehouse_id:
+                continue
+            if rec.dock_id and rec.dock_id.warehouse_id != rec.warehouse_id:
```

Comment:
It can be simple like this
```suggestion
            if rec.dock_id.warehouse_id != rec.warehouse_id:
```